python/lsst/summit/utils/astrometry/plotting.py

```diff
@@ -0,0 +1,133 @@
+
+    Plots the exposure on a logNorm scale, with the brightest sources, as
+    selected by the configuration, overlaid with an x. Plots compass arrows
+    for both north/east and az/el. Optionally saves the output to a file is
+    ``saveAs`` is supplied.
+
+    Parameters
+    ----------
+    exp : `lsst.afw.image.Exposure`
+        The exposure to get the astrometry for.
+    icSrc : `lsst.afw.table.SourceCatalog`
+        The source catalog for the exposure.
+    filteredSources : `lsst.afw.table.SourceCatalog`, optional
+        The filtered source catalog. If supplied, shows which sources were
+        selected.
+    saveAs : `str`, optional
+        Saves the plot to this filename if specified.
+    """
+    plt.figure(figsize=(16, 16))
+    arr = exp.image.array
+    arr = np.clip(arr, 1, 1000000)
+    arr = quickSmooth(arr)  # smooth slightly to help visualize
+    plt.imshow(np.arcsinh(arr)/10,
+               interpolation='None', cmap='gray', origin='lower')
+
+    height, width = exp.image.array.shape
+    leftFraction = .15  # fraction into the image to start the N/E compass
+    rightFraction = .225  # fraction into the image to start the az/el compass
+    fontsize = 20  # for the compass labels
+    compassSize = 500
+    textDistance = 650
+    compassCenter = (leftFraction*width, leftFraction*height)
+    compassAzElCent = ((1-rightFraction)*width, rightFraction*height)
+
+    vi = exp.getInfo().getVisitInfo()
+    az, _ = vi.boresightAzAlt
+    _, dec = vi.boresightRaDec
+    rotpa = vi.boresightRotAngle
+
+    az = Angle(az.asDegrees(), u.deg)
+    dec = Angle(dec.asDegrees(), u.deg)
+    rotpa = Angle(rotpa.asDegrees(), u.deg)
+
+    if icSrc:
+        plt.scatter(icSrc['base_SdssCentroid_x'], icSrc['base_SdssCentroid_y'], color='red', marker='x')
+    if filteredSources:
+        markerStyle = dict(marker='o', linestyle='', markersize=20, linewidth=10, color='green',
+                           markeredgecolor='green', fillstyle='none')
+        plt.plot(filteredSources['base_SdssCentroid_x'],
+                 filteredSources['base_SdssCentroid_y'],
+                 **markerStyle)
+    plt.arrow(compassCenter[0],
+              compassCenter[1],
+              -compassSize*np.sin(rotpa),
+              compassSize*np.cos(rotpa),
+              color='green', width=20)
+    plt.text(compassCenter[0]-textDistance*np.sin(rotpa),
+             compassCenter[1]+textDistance*np.cos(rotpa),
+             'N',
+             color='green', fontsize=fontsize, weight='bold')
+    plt.arrow(compassCenter[0],
+              compassCenter[1],
+              compassSize*np.cos(rotpa),
+              compassSize*np.sin(rotpa),
+              color='green', width=20)
+    plt.text(compassCenter[0]+textDistance*np.cos(rotpa),
+             compassCenter[1]+textDistance*np.sin(rotpa),
+             'E',
+             color='green', fontsize=fontsize, weight='bold')
+
+    sinTheta = np.cos(AUXTEL_LOCATION.lat) / np.cos(dec) * np.sin(az)
```

Comment:
DM style guide says no spaces around `/` and `*` (I don't really care which convention you pick but do be consistent as there are no spaces below).